### PR TITLE
fix: PowerMut incorrect precision

### PR DESCRIPTION
**Pull request by p0mvn**
Closes: #XXX

## What is the purpose of the change

Optimization in https://github.com/osmosis-labs/cosmos-sdk/pull/443 broke an edge case.

The earlier change would initialize one to incorrect precision. This fixes the bug. Tested on the osmosis that all test cases pass.

### CODE REVIEW

**Review by czarcas7ic — APPROVED** [review]
ACK

File: types/decimal.go [hunk 1]
```diff
@@ -499,8 +499,8 @@ func (d Dec) Power(power uint64) Dec {
 func (d Dec) PowerMut(power uint64) Dec {
 	// TODO: use mutable functions here
 	if power == 0 {
-		d.i.SetInt64(1)
-		return d
+		// Set to 1 with the correct precision.
+		d.i.Set(precisionReuse)
```
Comment (ValarDragon) [inline types/decimal.go:503]: oof yeah
